client/lib/query-manager/index.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * External dependencies
+ */
+import isEqual from 'lodash/isEqual';
+import reduce from 'lodash/reduce';
+import values from 'lodash/values';
+import omit from 'lodash/omit';
+import map from 'lodash/map';
+
+/**
+ * Internal dependencies
+ */
+import QueryKey from './key';
+
+/**
+ * QueryManager manages items which can be queried and change over time. It is
+ * intended to be extended by a more specific implementation which is
+ * responsible for implementing its matching, merging, and sorting behaviors.
+ */
+export default class QueryManager {
+	/**
+	 * Returns an instance of QueryManager or extending class given a
+	 * serialized string representation of the instance. If the serialized
+	 * string is invalid, a new empty instance is returned.
+	 *
+	 * @param  {String}       serialized Serialized QueryManager
+	 * @return {QueryManager}            QueryManager instance
+	 */
+	static parse( serialized ) {
+		try {
+			const { data, options } = JSON.parse( serialized );
+			return new this.prototype.constructor( data, options );
+		} catch ( e ) {
+			return new this.prototype.constructor();
+		}
+	}
+
+	/**
+	 * Constructs a new instance of QueryManager
+	 *
+	 * @param {Object} data    Initial data
+	 * @param {Object} options Manager options
```

Comment:
>It'd be helpful here to list out the option properties and what they do.
 
Added options property documentation to constructor and `receive` functions in rebased 25a39e2833b1e41f08dad263ec2620e032e4a308 ([line](https://github.com/Automattic/wp-calypso/pull/5135/commits/25a39e2833b1e41f08dad263ec2620e032e4a308#diff-90a76f3c723777b1a0c90ea618099ae8R38)).